crates/task-impls/src/transactions.rs

```diff
@@ -219,30 +219,15 @@ where
                 // TODO (Keyao) Determine whether to allow empty blocks.
                 // <https://github.com/EspressoSystems/HotShot/issues/1822>
                 let txns = self.wait_for_transactions(parent_leaf).await?;
+                let encoded_txns = VIDTransaction::encode(txns.clone());
```

Comment:
My understanding is that here we should instead call `build(txns)` for some generic `BlockPayload`. I don't think the struct `VIDTransaction` should exist in production code.